src/main/java/net/dv8tion/jda/api/entities/IPermissionHolder.java

```diff
@@ -26,8 +26,57 @@
  *
  * @since 3.0
  */
-public interface IPermissionHolder
+public interface IPermissionHolder extends ISnowflake
 {
+    /**
+     * Converts this abstraction to the concrete {@link net.dv8tion.jda.api.entities.Role Role} type.
+     * <br>If this is not a role, this will return {@code null} instead.
+     *
+     * @return This permission holder as a concrete Role, or null
+     *
+     * @see    #isRole()
+     * @see    #asMember()
+     */
+    default Role asRole()
+    {
+        return isRole() ? (Role) this : null;
+    }
+
+    /**
+     * Converts this abstraction to the concrete {@link net.dv8tion.jda.api.entities.Member Member} type.
+     * <br>If this is not a member, this will return {@code null} instead.
+     *
+     * @return This permission holder as a concrete Member, or null
+     *
+     * @see    #isMember()
+     * @see    #asRole()
+     */
+    default Member asMember()
```

Comment:
This means that `Role#asMember()` is a thing now, yes?

Feel like this is an anti-pattern of some kind, but I've nothing more to say on the topic.